### ISSUE TITLE
Include `variables` in query props

### ISSUE THREAD

**Issue by simenbrekken** [apollostack/react-apollo#71]
There's currently no way to access variables inside a component wrapped with `connect`.

It would be great if `variables` passed to queries in `connect()` were also passed as a query prop.

E.g:

```javascript

import React from 'react'

import gql from 'apollo-client/gql'
import { connect } from 'react-apollo'

const MyComponent = { myQuery } => {
  const { items = [], variables, loading, refetch } = myQuery
  const loadMore = () => refetch({
    ...variables,
    skip: variables.skip + variables.limit
  })

  return (
    <div>
      {items.map(item => <div key={item.id}>{item.text}</div>)}

      <button onClick={loadMore}>Load more</button>
    </div>
  )
}

export default connect({
  mapQueriesToProps() {
    return {
      myQuery: {
        query: gql`
          query (
            $limit: Int
            $skip: String
          ) {
            items(
              limit: $limit
              skip: $skip
            ) {
              id
              text
            }
          }
        `,
        variables: {
          limit: 10,
          skip: 0,
        },
      },
    }
  },
})(MyComponent)
```

**Comment by helfer** [comment]
@simenbrekken Is there a good reason for not setting the variables through props? In my opinion variables should always be set through props or state, otherwise your components are going to be difficult to understand and maintain.

**Comment by jbaxleyiii** [comment]
@simenbrekken I think this could be valuable for merging / using original props in refetch methods. I'll add it in the 1.0

**Comment by mukkachaitanya** [comment]
I have been looking around and couldn't find the way to pass the variables through the components state. Can anyone help me with that? 
Thanks

**Comment by ElNinjaGaiden** [comment]
Anyone with an idea how to pass the variables through the components state?? I'm stuck with the same problem than @mukkachaitanya

**Comment by kevinlu1211** [comment]
I don't know if I'm understanding the question correctly, but you can set the variables through the props by doing something like this:

The container component:

```
class Foo extends React.Component {
  render() {
    const { results } = this.props;
    const { jobId } = this.state;
    return (
      <div>
        <ResultList jobId={jobId}>
      </div>
    );
  }
}
```

The HOC displaying the data:

```
class ResultsList extends React.Component {
  render() {
    const { data: { loading, error, results } } = this.props;
    if (loading) {
      return <p>Loading ...</p>;
    }
    if (error) {
      return <p>{error.message}</p>;
    }
    return (
      <div className="resultList">
        {results.map(r => <div key={r._id}>{r._id}</div>)}
      </div>
    );
  }
}

export const resultsListQuery = gql`
  query ResultListQuery($jobId: jobId) {
    results(jobId: $jobId) {
      _id
    }
  }
`;

export default graphql(resultsListQuery, {
  options: props => ({
    variables: { jobId: props.jobId }, // use the props here
    name: "results",
    pollInterval: 2500
  })
})(ResultsList);
```

**Comment by Aspintyl000** [comment]
@ElNinjaGaiden @mukkachaitanya 

The way I did it (Probably a terrible idea) is to use Redux for a single app wide state and then store the variable I want to change in the Redux Store. Then when I had a component that needed a component a prop variable I would pass it the Redux state reference. Then inside the component I gave it access to the action to change the the Redux state. If that makes sense.
